File: src/structure_and_interpretation_of_computer_programs/other/pascal_triangle.py
def pascal(row: int, col: int) -> int:
    if row == 1:
        return 1
    if col == 1 or col == row:
        return 1
    else:
        return pascal(row - 1, col - 1) + pascal(row - 1, col)


if __name__ == "__main__":
    for row in range(1, 10):
        res = []
        for col in range(1, row + 1):
            res.append(pascal(row, col))
        print(res)

File: src/structure_and_interpretation_of_computer_programs/abstraction_with_procedures/pascal_triangle.clj
(defn pascal [row, col]
  (cond
    (= row 1) 1
    (or (= col 1) (= col row)) 1
    :else (+ (pascal (- row 1) (- col 1)) (pascal (- row 1) col))))

(defn display-pascal-row [n]
  (defn column-iter [i]
    (print (pascal n i)) (print "  ")
    (if (= i n)
      (newline)
      (column-iter (+ i 1))))
  (column-iter 1))

(defn display-pascal [n]
  (defn display-pascal-iter [i]
    (display-pascal-row i)
    (if (= i n)
      (newline)
      (display-pascal-iter (+ i 1))))
  (display-pascal-iter 1))

(display-pascal 14)